### ISSUE TITLE
Endpoint is not ready in istio test

### ISSUE THREAD

**Issue by eloycoto** [cilium/cilium#5859]
***Summary***: Endpoint on Istio test does not get in ready state after timeout

Fail
```
home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:383
Endpoints are not ready after timeout
Expected
    <*helpers.SSHMetaError | 0xc424e670e0>: Error: Timeout reached: Endpoints are not ready after timeout
Extended info: 	Cilium Pod: cilium-5dvq5 	Endpoint: 3154 	Identity: 4	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 10271 	Identity: 27690	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 23570 	Identity: 62332	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 28293 	Identity: 128	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 30015 	Identity: 128	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 30128 	Identity: 129	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 30540 	Identity: 21151	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 33196 	Identity: 37219	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 62365 	Identity: 35029	 State: ready
	Cilium Pod: cilium-5dvq5 	Endpoint: 64728 	Identity: 29638	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 8465 	Identity: 7130	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 8570 	Identity: 40569	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 10008 	Identity: 43158	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 16757 	Identity: 44364	 State: not-ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 20133 	Identity: 128	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 22232 	Identity: 63832	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 24347 	Identity: 28966	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 28818 	Identity: 32336	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 30313 	Identity: 29492	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 33009 	Identity: 128	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 39371 	Identity: 46782	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 42140 	Identity: 4	 State: ready
	Cilium Pod: cilium-ssdw9 	Endpoint: 64117 	Identity: 11734	 State: ready
```

Full output: 
```
Number of 'context deadline exceeded' in logs: 4
⚠️  Number of 'level=error' in logs: 8
⚠️  Number of 'level=warning' in logs: 11
Number of 'Cilium API handler panicked' in logs: 0
⚠️  Number of 'Goroutine took lock for more than' in logs: 115
Cilium pods: [cilium-5dvq5 cilium-ssdw9]
Netpols loaded: 
CiliumNetworkPolicies loaded: default::cnp-second-namespace default::cnp-specs default::istio-sidecar 
Endpoint Policy Enforcement:
     cilium-etcd-bskkvd6dtz                      =>   none
     istio-galley-5664cf4486-thmwz               =>   none
     istio-telemetry-5c7b4697dd-9z8qk            =>   none
     istio-sidecar-injector-7ddfb55469-2cdbx     =>   none
     cilium-etcd-98nkhhwqnm                      =>   none
     etcd-operator-7cb79cdf99-f4fjq              =>   none
     istio-egressgateway-7d4f6d4ccd-lbsv5        =>   none
     istio-policy-746f846dd5-4bsq9               =>   none
     cilium-etcd-s2g7cd7j47                      =>   none
     prometheus-core-5888d94764-vb6q2            =>   none
     details-v1-5f85dc9586-nb49h                 =>   ingress
     reviews-v2-57456d49fb-tl55p                 =>   none
     reviews-v1-769dc8c876-86snt                 =>   none
     istio-citadel-657777464f-vmk4w              =>   none
     istio-ingressgateway-5c4cf57d59-wkbpv       =>   none
     istio-pilot-84bc576c97-dtwx4                =>   none
     istio-statsd-prom-bridge-67bbcc746c-k8fmk   =>   none
     prometheus-6967c997cf-4llnw                 =>   none
     productpage-v1-5bdbc7dbb8-7tkgw             =>   none
     ratings-v1-85c7c99486-7cfrc                 =>   ingress
     coredns-65b788cf-tsjks                      =>   none
Cilium agent 'cilium-5dvq5': Status: Ok  Health: Ok Nodes "k8s2 k8s1" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 78 Failed 0
Cilium agent 'cilium-ssdw9': Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 89 Failed 0

Standard Error

STEP: Creating the istio-system namespace
STEP: Creating the Istio resources
STEP: Waiting for Istio pods to be ready
STEP: Waiting for Istio service "istio-statsd-prom-bridge" to be ready
STEP: Waiting for Istio service "istio-pilot" to be ready
STEP: Waiting for DNS to resolve Istio service "istio-statsd-prom-bridge"
STEP: Waiting for DNS to resolve Istio service "istio-pilot"
STEP: Creating policy in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/istio-sidecar-init-policy.yaml"
STEP: Creating policy in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/cnp-specs.yaml"
STEP: Creating resources in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/bookinfo-v2-istio.yaml"
STEP: Creating resources in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/bookinfo-v1-istio.yaml"
STEP: Waiting for Bookinfo pods to be ready
STEP: Waiting for Bookinfo endpoints to be ready
=== Test Finished at 2018-10-11T07:38:14Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
Exitcode: 0 
Stdout:
 	 NAMESPACE      NAME                                        READY   STATUS      RESTARTS   AGE     IP              NODE   NOMINATED NODE
	 default        details-v1-5f85dc9586-nb49h                 2/2     Running     1          8m34s   10.10.0.124     k8s1   <none>
	 default        productpage-v1-5bdbc7dbb8-7tkgw             2/2     Running     0          8m34s   10.10.1.154     k8s2   <none>
	 default        ratings-v1-85c7c99486-7cfrc                 2/2     Running     0          8m35s   10.10.0.50      k8s1   <none>
	 default        reviews-v1-769dc8c876-86snt                 2/2     Running     0          8m34s   10.10.1.104     k8s2   <none>
	 default        reviews-v2-57456d49fb-tl55p                 2/2     Running     0          8m35s   10.10.0.74      k8s1   <none>
	 istio-system   istio-citadel-657777464f-vmk4w              1/1     Running     0          11m     10.10.0.113     k8s1   <none>
	 istio-system   istio-cleanup-secrets-2tgxx                 0/1     Completed   0          11m     10.10.1.87      k8s2   <none>
	 istio-system   istio-egressgateway-7d4f6d4ccd-lbsv5        1/1     Running     0          11m     10.10.1.77      k8s2   <none>
	 istio-system   istio-galley-5664cf4486-thmwz               1/1     Running     0          11m     10.10.0.77      k8s1   <none>
	 istio-system   istio-ingressgateway-5c4cf57d59-wkbpv       1/1     Running     0          11m     10.10.1.172     k8s2   <none>
	 istio-system   istio-pilot-84bc576c97-dtwx4                2/2     Running     0          11m     10.10.1.204     k8s2   <none>
	 istio-system   istio-policy-746f846dd5-4bsq9               2/2     Running     0          11m     10.10.1.250     k8s2   <none>
	 istio-system   istio-security-post-install-8j9qv           0/1     Completed   5          11m     10.10.0.49      k8s1   <none>
	 istio-system   istio-sidecar-injector-7ddfb55469-2cdbx     1/1     Running     0          11m     10.10.0.153     k8s1   <none>
	 istio-system   istio-statsd-prom-bridge-67bbcc746c-k8fmk   1/1     Running     0          11m     10.10.0.162     k8s1   <none>
	 istio-system   istio-telemetry-5c7b4697dd-9z8qk            2/2     Running     0          11m     10.10.0.223     k8s1   <none>
	 istio-system   prometheus-6967c997cf-4llnw                 1/1     Running     0          11m     10.10.0.22      k8s1   <none>
	 kube-system    cilium-5dvq5                                1/1     Running     0          57m     192.168.36.12   k8s2   <none>
	 kube-system    cilium-etcd-98nkhhwqnm                      1/1     Running     0          57m     10.10.0.238     k8s1   <none>
	 kube-system    cilium-etcd-bskkvd6dtz                      1/1     Running     0          56m     10.10.0.189     k8s1   <none>
	 kube-system    cilium-etcd-s2g7cd7j47                      1/1     Running     0          57m     10.10.1.248     k8s2   <none>
	 kube-system    cilium-ssdw9                                1/1     Running     0          57m     192.168.36.11   k8s1   <none>
	 kube-system    coredns-65b788cf-tsjks                      1/1     Running     0          58m     10.10.1.80      k8s2   <none>
	 kube-system    etcd-k8s1                                   1/1     Running     0          84m     192.168.36.11   k8s1   <none>
	 kube-system    etcd-operator-7cb79cdf99-f4fjq              1/1     Running     0          58m     10.10.1.197     k8s2   <none>
	 kube-system    kube-apiserver-k8s1                         1/1     Running     0          84m     192.168.36.11   k8s1   <none>
	 kube-system    kube-controller-manager-k8s1                1/1     Running     0          84m     192.168.36.11   k8s1   <none>
	 kube-system    kube-proxy-cwsx9                            1/1     Running     0          85m     192.168.36.11   k8s1   <none>
	 kube-system    kube-proxy-nlw5k                            1/1     Running     0          78m     192.168.36.12   k8s2   <none>
	 kube-system    kube-scheduler-k8s1                         1/1     Running     0          84m     192.168.36.11   k8s1   <none>
	 monitoring     prometheus-core-5888d94764-vb6q2            1/1     Running     0          77m     10.10.1.24      k8s2   <none>
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-5dvq5 -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                                                    IPv6                   IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                                      
	 3154       Disabled           Disabled          4          reserved:health                                                                f00d::a0a:100:0:c52    10.10.1.42    ready   
	 10271      Disabled           Disabled          27690      k8s:app=istio-ingressgateway                                                   f00d::a0a:100:0:281f   10.10.1.172   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-ingressgateway-service-account                                                
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                
	                                                            k8s:istio=ingressgateway                                                                                                    
	 23570      Disabled           Disabled          62332      k8s:app=policy                                                                 f00d::a0a:100:0:5c12   10.10.1.250   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-mixer-service-account                                                         
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                
	                                                            k8s:istio-mixer-type=policy                                                                                                 
	                                                            k8s:istio=mixer                                                                                                             
	 28293      Disabled           Disabled          128        container:annotation.etcd.version=3.3.9                                        f00d::a0a:100:0:6e85   10.10.1.248   ready   
	                                                            k8s:app=etcd                                                                                                                
	                                                            k8s:etcd_cluster=cilium-etcd                                                                                                
	                                                            k8s:etcd_node=cilium-etcd-s2g7cd7j47                                                                                        
	                                                            k8s:io.cilium.fixed-identity=kv-store                                                                                       
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                             
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                                                 
	 30015      Disabled           Disabled          128        k8s:io.cilium.fixed-identity=kv-store                                          f00d::a0a:100:0:753f   10.10.1.197   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                                                      
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                                                 
	                                                            k8s:name=etcd-operator                                                                                                      
	 30128      Disabled           Disabled          129        k8s:io.cilium.fixed-identity=kube-dns                                          f00d::a0a:100:0:75b0   10.10.1.80    ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                                             
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                                                 
	                                                            k8s:k8s-app=kube-dns                                                                                                        
	 30540      Disabled           Disabled          21151      k8s:app=istio-egressgateway                                                    f00d::a0a:100:0:774c   10.10.1.77    ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-egressgateway-service-account                                                 
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                
	                                                            k8s:istio=egressgateway                                                                                                     
	 33196      Disabled           Disabled          37219      k8s:app=productpage                                                            f00d::a0a:100:0:81ac   10.10.1.154   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.istiosidecarproxy=true                                                                             
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                             
	                                                            k8s:io.kubernetes.pod.namespace=default                                                                                     
	                                                            k8s:track=stable                                                                                                            
	                                                            k8s:version=v1                                                                                                              
	                                                            k8s:zgroup=bookinfo                                                                                                         
	 62365      Disabled           Disabled          35029      k8s:app=pilot                                                                  f00d::a0a:100:0:f39d   10.10.1.204   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-pilot-service-account                                                         
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                
	                                                            k8s:istio=pilot                                                                                                             
	 64728      Disabled           Disabled          29638      k8s:app=reviews                                                                f00d::a0a:100:0:fcd8   10.10.1.104   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                    
	                                                            k8s:io.cilium.k8s.policy.istiosidecarproxy=true                                                                             
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                             
	                                                            k8s:io.kubernetes.pod.namespace=default                                                                                     
	                                                            k8s:track=stable                                                                                                            
	                                                            k8s:version=v1                                                                                                              
	                                                            k8s:zgroup=bookinfo                                                                                                         
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-5dvq5 -- cilium bpf proxy list
Exitcode: 0 
Stdout:
 	 KEY   VALUE   
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-ssdw9 -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                                                      IPv6                 IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                                      
	 8465       Disabled           Disabled          7130       k8s:app=reviews                                                                  f00d::a0a:0:0:2111   10.10.0.74    ready       
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                        
	                                                            k8s:io.cilium.k8s.policy.istiosidecarproxy=true                                                                                 
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                 
	                                                            k8s:io.kubernetes.pod.namespace=default                                                                                         
	                                                            k8s:version=v2                                                                                                                  
	                                                            k8s:zgroup=bookinfo                                                                                                             
	 8570       Disabled           Disabled          40569      k8s:io.cilium.k8s.policy.cluster=default                                         f00d::a0a:0:0:217a   10.10.0.77    ready       
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-galley-service-account                                                            
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                    
	                                                            k8s:istio=galley                                                                                                                
	 10008      Disabled           Disabled          43158      k8s:io.cilium.k8s.policy.cluster=default                                         f00d::a0a:0:0:2718   10.10.0.162   ready       
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-mixer-service-account                                                             
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                    
	                                                            k8s:istio=statsd-prom-bridge                                                                                                    
	 16757      Enabled            Disabled          44364      k8s:app=details                                                                  f00d::a0a:0:0:4175   10.10.0.124   not-ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                        
	                                                            k8s:io.cilium.k8s.policy.istiosidecarproxy=true                                                                                 
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                 
	                                                            k8s:io.kubernetes.pod.namespace=default                                                                                         
	                                                            k8s:track=stable                                                                                                                
	                                                            k8s:version=v1                                                                                                                  
	                                                            k8s:zgroup=bookinfo                                                                                                             
	 20133      Disabled           Disabled          128        container:annotation.etcd.version=3.3.9                                          f00d::a0a:0:0:4ea5   10.10.0.189   ready       
	                                                            k8s:app=etcd                                                                                                                    
	                                                            k8s:etcd_cluster=cilium-etcd                                                                                                    
	                                                            k8s:etcd_node=cilium-etcd-bskkvd6dtz                                                                                            
	                                                            k8s:io.cilium.fixed-identity=kv-store                                                                                           
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                        
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                 
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                                                     
	 22232      Disabled           Disabled          63832      k8s:app=prometheus                                                               f00d::a0a:0:0:56d8   10.10.0.22    ready       
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                        
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=prometheus                                                                              
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                    
	 28818      Enabled            Disabled          32336      k8s:app=ratings                                                                  f00d::a0a:0:0:7092   10.10.0.50    ready       
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                        
	                                                            k8s:io.cilium.k8s.policy.istiosidecarproxy=true                                                                                 
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                 
	                                                            k8s:io.kubernetes.pod.namespace=default                                                                                         
	                                                            k8s:version=v1                                                                                                                  
	                                                            k8s:zgroup=bookinfo                                                                                                             
	 30313      Disabled           Disabled          29492      k8s:app=telemetry                                                                f00d::a0a:0:0:7669   10.10.0.223   ready       
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                        
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-mixer-service-account                                                             
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                    
	                                                            k8s:istio-mixer-type=telemetry                                                                                                  
	                                                            k8s:istio=mixer                                                                                                                 
	 33009      Disabled           Disabled          128        container:annotation.etcd.version=3.3.9                                          f00d::a0a:0:0:80f1   10.10.0.238   ready       
	                                                            k8s:app=etcd                                                                                                                    
	                                                            k8s:etcd_cluster=cilium-etcd                                                                                                    
	                                                            k8s:etcd_node=cilium-etcd-98nkhhwqnm                                                                                            
	                                                            k8s:io.cilium.fixed-identity=kv-store                                                                                           
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                                                        
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                 
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                                                     
	 39371      Disabled           Disabled          46782      k8s:io.cilium.k8s.policy.cluster=default                                         f00d::a0a:0:0:99cb   10.10.0.113   ready       
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-citadel-service-account                                                           
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                    
	                                                            k8s:istio=citadel                                                                                                               
	 42140      Disabled           Disabled          4          reserved:health                                                                  f00d::a0a:0:0:a49c   10.10.0.242   ready       
	 64117      Disabled           Disabled          11734      k8s:io.cilium.k8s.policy.cluster=default                                         f00d::a0a:0:0:fa75   10.10.0.153   ready       
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=istio-sidecar-injector-service-account                                                  
	                                                            k8s:io.kubernetes.pod.namespace=istio-system                                                                                    
	                                                            k8s:istio=sidecar-injector                                                                                                      
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-ssdw9 -- cilium bpf proxy list
Exitcode: 0 
Stdout:
 	 KEY   VALUE   
	 
Stderr:
 	 

===================== Existing AfterFailed =====================
STEP: Deleting resource in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/bookinfo-v2-istio.yaml"
STEP: Deleting resource in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/bookinfo-v1-istio.yaml"
STEP: Deleting policy in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/istio-sidecar-init-policy.yaml"
STEP: Deleting policy in file "/home/vagrant/go/src/github.com/cilium/cilium/test/k8sT/manifests/cnp-specs.yaml"
STEP: Deleting the Istio resources
STEP: Deleting the istio-system namespace

[[ATTACHMENT|75421305_K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter-service_connectivity.zip]]
```

Link: https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/4368/testReport/k8s-1/12/K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter_service_connectivity/
Logs: 
[75421305_K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter-service_connectivity.zip](https://github.com/cilium/cilium/files/2468228/75421305_K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter-service_connectivity.zip)

**Comment by rlenglet** [comment]
The problem is that in some cases, the `istio-proxy` container starts before the pod's BPF program is installed, i.e. before it has L3/L4 connectivity.

The `istio-proxy` container immediately tries to resolve some DNS names after starting. There seems to be a bug in the DNS resolver within the `istio-proxy` container: if it fails to resolve a name the first time because it can't connect to the DNS server, it will never be able to resolve it again until the container dies. Even it if keeps retrying for more than 3 minutes, long after the BPF program is installed and the pod actually has L3/L4 connectivity. This is not an issue in Envoy, since Envoy is restarted several times within those 3 minutes. And this isn't an issue in Cilium. That seems to be a problem with the container's system resolver.

The reason the pod can start before the BPF program is installed is that we shortcut the synchronous wait for `ready` state on endpoint creation in the case the endpoint is Istio-injected, to avoid the deadlock with Envoy configuration in case the endpoint has a policy with HTTP rules.
We return from a synchronous endpoint creation API call in one of two cases:

1. The endpoint is `ready`.
2. The endpoint has received an identity, and has been determined to be Istio-injected.

We are in case 2. here. And in this case, this return from the endpoint creation API call can happen before the BPF program is installed.

This issue doesn't happen often in this test, because every pod is injected with an init container that sleeps for 30 seconds. That reduces the chance that the `istio-proxy` container starts before the BPF program is installed. But that doesn't completely eliminate the probability.

I can see 2 ways to fix this:

1. Return from endpoint creation if the endpoint has received an identity, has been determined to be Istio-injected **AND has a BPF program installed**.
2. Replace the init container to wait for the Istio DNS names to be resolvable, instead of just sleeping for 30 seconds.

**Comment by tgraf** [comment]
I think option 1 b) is the only reasonable fix for this.

**Comment by eloycoto** [comment]
I disabled the istio test in master,  please revert the commit when this is fixed. 

Thanks!

**Comment by ianvernon** [comment]
Reopening as #5869, while a good addition, isn’t a true fix to this.

**Comment by jrajahalme** [comment]
Still see endpoints in not-ready state running the Istio GSG locally. It seems that the istio_proxy container takes a long time to start as it remains in ContainerInitializing state for longer than the regeneration timeout. This happens especially when the container images need to be pulled.

Noticed that the not-ready endpoints can be fixed by forcing regeneration (e.g., by flipping the debug config option), which indicates that the error was only due to a timeout rather than a more functional reason.

Testing a fix in #6211, which allows for longer regeneration timeouts. The side-effects of longer timeouts are mitigated by allowing other endpoints to build by releasing the "build permit" after the BPF program has been compiled, before starting the Wait for the proxy ACKs.

**Comment by ianvernon** [comment]
Finally able to close this; this has been quite a wild ride. 

To summarize the timeline of what happened in case anyone is interested:

* Due to extreme load on the CI machines from the amount of pods running in this test, as well as from microscope being unable to handle load from all of the flows in this test, we discovered an issue in which identity propagation would not occur quickly enough, causing packet drops on ingress to pods which allow traffic from all endpoints. This was due to the fact that allow-all equates to allowing traffic to / from all identities, but if an identity has not yet been plumbed into the policy map for an endpoint, even though no policy selected that endpoint, the packets would be dropped. This was fixed by 
https://github.com/cilium/cilium/pull/6208 . We also disabled microscope in the CI as it has scale issues.
* @jrajahalme optimized the code to allow for other endpoint builds to occur while waiting for proxy acks, instead of the entire regeneration process being complete in https://github.com/cilium/cilium/pull/6211 . An increase in build timeouts also reduced the propensity for endpoints going into `not-ready` state.
* The removal of the init identity resulted in any pod injected with an istio-sidecar proxy never being able to regenerate, due to the issue documented here: 
https://github.com/cilium/cilium/pull/6467#issuecomment-451282748 . The fix for this was to remove synchronous regeneration from the `createEndpoint` function, and instead rely on signals via channels for when BPF of an endpoint is compiled, as opposed to waiting for the entire regeneration process, including L7 proxy configuration, to be completed. This helps avoid chicken-and-egg issues with the istio-proxy and Cilium.
* Finally, 
https://github.com/cilium/cilium/pull/6467 re-enabled this test, and the test has proven stable across many CI runs. The test is now stable. Closing this.
